app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java

```diff
@@ -532,10 +532,19 @@ private LineData generateKetonesData() {
     private LineData generateWeightData() {
         List<String> xVals = new ArrayList<>();
         List<Entry> yVals = new ArrayList<>();
+        boolean needUnitConversion = false;
+
+        if (!"kilograms".equals(presenter.getWeightUnitMeasuerement())) {
+            needUnitConversion = true;
+        }
 
         int k = 0;
         for (int i = presenter.getWeightReadings().size() - 1; i >= 0; i--) {
             float val = Float.parseFloat(presenter.getWeightReadings().get(i).toString());
+            if (needUnitConversion) {
+                val = (float)GlucosioConverter.kgToLb(val);
```

Comment:
Was it forgotten?

Reply:
The conversion I mean

Reply:
I don't understand this comment

Reply:
Question was - we never had this functionality so we never had proper working weight for imperial countries, correct?

Reply:
Correct, as far as I can tell it never worked for imperial countries

Reply:
👍